cpp/arcticdb/storage/azure/azure_storage.cpp

```diff
@@ -61,27 +61,41 @@ std::string get_error_code(const Azure::Core::RequestFailedException& e) {
     return error_code;
 }
 
-void raise_azure_exception(const Azure::Core::RequestFailedException& e) {
+void raise_azure_exception(const Azure::Core::RequestFailedException& e, const std::string& object_name) {
     auto error_code = get_error_code(e);
     auto status_code = e.StatusCode;
     std::string error_message;
 
     if(status_code == Azure::Core::Http::HttpStatusCode::NotFound && error_code == AzureErrorCode_to_string(AzureErrorCode::BlobNotFound)) {
-        throw KeyNotFoundException(fmt::format("Key Not Found Error: AzureError#{} {}: {}",
-                                               static_cast<int>(status_code), error_code, e.ReasonPhrase));
+        throw KeyNotFoundException(fmt::format("Key Not Found Error: AzureError#{} {}: {} for object {}",
+                                               static_cast<int>(status_code),
+                                               error_code, 
+                                               e.ReasonPhrase,
+                                               object_name));
     }
 
     if(status_code == Azure::Core::Http::HttpStatusCode::Unauthorized || status_code == Azure::Core::Http::HttpStatusCode::Forbidden) {
-        raise<ErrorCode::E_PERMISSION>(fmt::format("Permission error: AzureError#{} {}: {}",
-                                                   static_cast<int>(status_code), error_code, e.ReasonPhrase));
+        raise<ErrorCode::E_PERMISSION>(fmt::format("Permission error: AzureError#{} {}: {} for object {}",
+                                                   static_cast<int>(status_code),
+                                                   error_code,
+                                                   e.ReasonPhrase,
+                                                   object_name));
     }
 
     if(static_cast<int>(status_code) >= 500) {
-        error_message = fmt::format("Unexpected Server Error: AzureError#{} {}: {} {}",
-                                    static_cast<int>(status_code), error_code, e.ReasonPhrase, e.what());
+        error_message = fmt::format("Unexpected Server Error: AzureError#{} {}: {} {} for object {}",
+                                    static_cast<int>(status_code),
+                                    error_code,
+                                    e.ReasonPhrase,
+                                    e.what(),
+                                    object_name);
     } else {
-        error_message = fmt::format("Unexpected Error: AzureError#{} {}: {} {}",
-                                    static_cast<int>(status_code), error_code, e.ReasonPhrase, e.what());
+        error_message = fmt::format("Unexpected Error: AzureError#{} {}: {} {} for object {}",
+                                    static_cast<int>(status_code),
+                                    error_code,
+                                    e.ReasonPhrase,
+                                    e.what(),
+                                    object_name);
```

Comment:
Could avoid some code duplication here by pulling this out:
```
error_message_suffix = fmt::format(AzureError#{} {}: {} {} for object {}",
                                    static_cast<int>(status_code),
                                    error_code,
                                    e.ReasonPhrase,
                                    e.what(),
                                    object_name);
```
Looks like the same is true on the other storages too